perun-core/src/test/java/cz/metacentrum/perun/core/entry/GroupsManagerEntryIntegrationTest.java

```diff
@@ -5251,6 +5251,55 @@ public void isGroupForAutoRegistration() throws Exception {
 		assertTrue(groupsManagerBl.isGroupForAutoRegistration(sess, group));
 	}
 
+	@Test
+	public void getGroupRichMembersByIds() throws Exception {
+		System.out.println(CLASS_NAME + "getGroupRichMembersByIds");
+
+		Vo vo = setUpVo();
+		Member member = setUpMember(vo);
+		Member member2 = setUpMember(vo);
+		groupsManagerBl.createGroup(sess, vo, group);
+		groupsManagerBl.addMember(sess, group, member);
+		groupsManagerBl.addMember(sess, group, member2);
+		User user = usersManagerBl.getUserByMember(sess, member);
+		User user2 = usersManagerBl.getUserByMember(sess, member2);
+
+		Attribute userAttribute1 = setUpAttribute(String.class.getName(), "testUserAttribute1", AttributesManager.NS_USER_ATTR_DEF, "TEST VALUE");
+		Attribute userAttribute2 = setUpAttribute(String.class.getName(), "testUserAttribute2", AttributesManager.NS_USER_ATTR_DEF, "TEST VALUE");
+		attributesManager.setAttributes(sess, user, new ArrayList<>(Arrays.asList(userAttribute1, userAttribute2)));
+		attributesManager.setAttributes(sess, user2, new ArrayList<>(Arrays.asList(userAttribute1, userAttribute2)));
+
+		Attribute memberAttribute1 = setUpAttribute(Integer.class.getName(), "testMemberAttribute1", AttributesManager.NS_MEMBER_ATTR_DEF, 15);
+		attributesManager.setAttribute(sess, member, memberAttribute1);
+		attributesManager.setAttribute(sess, member2, memberAttribute1);
+
+		List<Attribute> memberGroupAttributes = setUpMemberGroupAttributes(); //sets 2 member-group attributes
+		perun.getAttributesManagerBl().setAttributes(sess, member, group, memberGroupAttributes);
+		perun.getAttributesManagerBl().setAttributes(sess, member2, group, memberGroupAttributes);
+
+		List<String> attrNames = List.of(userAttribute1, memberAttribute1, memberGroupAttributes.get(0), memberGroupAttributes.get(1)).stream()
+			.map(AttributeDefinition::getName)
+			.collect(Collectors.toList());
+
+		List<RichMember> richMembers = groupsManager.getGroupRichMembersByIds(sess,  group.getId(), List.of(member.getId(), member.getId() + 10), attrNames);
+
+		//non-existing and not requested members are silently skipped
+		assertTrue(richMembers.size() == 1);
+		assertTrue(richMembers.get(0).getId() == member.getId());
+
+		List<Attribute> userAttributes = richMembers.get(0).getUserAttributes();
+		List<Attribute> memberAttributes = richMembers.get(0).getMemberAttributes();
+
+		//only requested user attribute is included
+		assertTrue(userAttributes.size() == 1);
+		assertTrue(userAttributes.contains(userAttribute1));
+		assertFalse(userAttributes.contains(userAttribute2));
+
+		//member and member-group attributes are included
+		assertTrue(memberAttributes.size() == 3);
+		assertTrue(memberAttributes.containsAll(List.of(memberAttribute1, memberGroupAttributes.get(0), memberGroupAttributes.get(1))));
+	}
```

Comment:
It is usually better to create multiple tests, each focusing on a single use case. It makes the code more readable and also the debugging. (e.g. if a test named `getGroupRichMembersByIds_returnsSpecifiedAttributes` fails, it is more clear what went wrong and is not working.)

Reply:
Split tests into separate cases.